Source/Example Measurements/NI-Digital SPI/NIDigitalSPI.lvproj

```diff
@@ -0,0 +1,126 @@
+				<Item Name="MeasurementLink Measurement Server.lvlib" Type="Library" URL="/&lt;vilib&gt;/MeasurementLink/MeasurementLink Measurement Server/MeasurementLink Measurement Server.lvlib"/>
+				<Item Name="MeasurementLink Measurement Service Base V2.lvlib" Type="Library" URL="/&lt;vilib&gt;/MeasurementLink/MeasurementLink Measurement Service Base V2/MeasurementLink Measurement Service Base V2.lvlib"/>
+				<Item Name="MeasurementLink niDigital.lvlib" Type="Library" URL="/&lt;vilib&gt;/MeasurementLink/MeasurementLink niDigital/MeasurementLink niDigital.lvlib"/>
+				<Item Name="MeasurementLink Pin Map Client.lvlib" Type="Library" URL="/&lt;vilib&gt;/MeasurementLink/MeasurementLink Pin Map Client/MeasurementLink Pin Map Client.lvlib"/>
+				<Item Name="MeasurementLink Session Management Client.lvlib" Type="Library" URL="/&lt;vilib&gt;/MeasurementLink/MeasurementLink Session Management Client/MeasurementLink Session Management Client.lvlib"/>
+				<Item Name="MeasurementLink Session Management.lvlib" Type="Library" URL="/&lt;vilib&gt;/MeasurementLink/MeasurementLink Session Management/MeasurementLink Session Management.lvlib"/>
+				<Item Name="NI_Data Type.lvlib" Type="Library" URL="/&lt;vilib&gt;/Utility/Data Type/NI_Data Type.lvlib"/>
+				<Item Name="NI_FileType.lvlib" Type="Library" URL="/&lt;vilib&gt;/Utility/lvfile.llb/NI_FileType.lvlib"/>
+				<Item Name="NI_LVConfig.lvlib" Type="Library" URL="/&lt;vilib&gt;/Utility/config.llb/NI_LVConfig.lvlib"/>
+				<Item Name="NI_PackedLibraryUtility.lvlib" Type="Library" URL="/&lt;vilib&gt;/Utility/LVLibp/NI_PackedLibraryUtility.lvlib"/>
+				<Item Name="Set Difference.vim" Type="VI" URL="/&lt;vilib&gt;/set operations/Set Difference.vim"/>
+				<Item Name="Space Constant.vi" Type="VI" URL="/&lt;vilib&gt;/dlg_ctls.llb/Space Constant.vi"/>
+				<Item Name="System Directory Type.ctl" Type="VI" URL="/&lt;vilib&gt;/Utility/sysdir.llb/System Directory Type.ctl"/>
+				<Item Name="System Exec.vi" Type="VI" URL="/&lt;vilib&gt;/Platform/system.llb/System Exec.vi"/>
+				<Item Name="Trim Whitespace.vi" Type="VI" URL="/&lt;vilib&gt;/Utility/error.llb/Trim Whitespace.vi"/>
+				<Item Name="whitespace.ctl" Type="VI" URL="/&lt;vilib&gt;/Utility/error.llb/whitespace.ctl"/>
+			</Item>
+			<Item Name="kernel32.dll" Type="Document" URL="kernel32.dll">
+				<Property Name="NI.PreserveRelativePath" Type="Bool">true</Property>
+			</Item>
+			<Item Name="niDigital_64.dll" Type="Document" URL="niDigital_64.dll">
+				<Property Name="NI.PreserveRelativePath" Type="Bool">true</Property>
+			</Item>
+			<Item Name="niDigitalLV_64.dll" Type="Document" URL="niDigitalLV_64.dll">
+				<Property Name="NI.PreserveRelativePath" Type="Bool">true</Property>
+			</Item>
+		</Item>
+		<Item Name="Build Specifications" Type="Build">
+			<Item Name="NI-Digital Measurement" Type="EXE">
```

Comment:
I'm not very familiar with building in LabVIEW. It wanted me to specify a startup project so I specified "Run Service.vi". It builds successfully now but running the exe doesn't work. It flickers the "Run Service.vi" and then disappears (crashes maybe? the process hangs and I have to kill it manually) and we can't connect to it. I saw the source files were included if referenced so I tried to explicitly include them in the build, but that didn't seem to make a difference. I also tried to recreate the Build Specification, which also didn't seem to make a difference. I also looked at some of the public documentation but didn't find much guidance there. I also double-checked that running "Run Service.vi" from the lvproj does work as expected.

 @pbirkhol-ni can you give some guidance on how I should fix this?

Reply:
Did this build specification come from instantiating this service from the template? Or did you create it by hand? If the latter, you may want to compare it to one from the template.

Reply:
Or from another example project.

Reply:
E.g. Cancellable measurement
![image](https://github.com/ni/measurementlink-labview/assets/38357562/16d8370b-6911-46ef-8681-0cc1d91b3f6c)

Reply:
Yeah, there are a bunch of always included VIs that are missing (VIs called via call by reference). I would start by copying from the template or one of the shipping examples and go from there. I'm not sure if buildspec configuration is persisted completely as XML in the lvproj file or if there is some binary data mixed in. If it's just XML, you could also try copy/pasting XML between projects by hand rather than relying on manual diffing to try and recreate everything. IIRC, I think I've seen mention of devs doing that in forums posts and generally being successful with it.

Reply:
I updated the lvproj to build similar to CancellableMeasurement. The main difference was the missing files under "Always Included" and missing the Post Build Action. I'm hitting an issue where the Post-Build Action.vi is failing due to a path issue (either relative or missing). 

Where do the "Post-Build Action.vi"'s inputs come from? I opened the VI expected it to populate when the build it run, but it does not. I also set a breakpoint in that VI, but it doesn't seem to get hit so I'm having trouble seeing what the issue is.

![image](https://github.com/ni/measurementlink-labview/assets/103216564/8e92cea2-f639-42d7-9f75-e202560cf763)

Reply:
I'm not sure where the inputs come from. It looks like maybe they are filled in automatically by convention from the build specification.

This code is copying some grpc stuff. Do you have the grpc-labview packages installed?
![image](https://github.com/ni/measurementlink-labview/assets/38357562/b4b5dd65-0ffa-46ef-82d8-7a204f55a9b1)

Reply:
Looks like this Generate VI button fills in the front panel, so I think these are provided by the application builder framework.
![Uploading image.png…]()

Reply:
The only things I see it copying is those grpc dependencies and the .measui file so check those two things.

Reply:
I have the grpc libraries installed and ensured that they are the latest versions from the latest release. I also had to copy over the latest `measurement-services-labview\Source\Runtime` into `vi.lib` for the latest session reservation stuff. 
![image](https://github.com/ni/measurementlink-labview/assets/103216564/d7952dc0-c1dd-46fb-b954-04b99645be12)

The .measui and .measproj files were removed as part of a previous commit and this example uses a LabVIEW UI now. Recreating them and adding them back in did not seem to make a difference. 

I noticed during the build process, I get this warning.
![image](https://github.com/ni/measurementlink-labview/assets/103216564/6d0715c4-e72d-446a-96c2-c7f4ef49e9af)

I checked and the `Measurement\_functions_ni_measurementlink_service_2.mnu` does not exist so this might be the cause. Where do these files come from? Maybe I need to reinstall MeasurementLink?

Reply:
I recently added a link to that .mnu file. The way it works is that the class is set up to point to a file that is not around during development time, but the VI package installs that file so it links up correctly to the class. I don't think that would cause this problem, but it's possible. Are other examples able to build a .exe correctly even though they get that warning?

Reply:
To debug and figure out where the error is coming from, you can go to the Post-Build Action.vi in vi.lib and put some error dialogs in there or an error handler on some various parts of the build process.

Reply:
Paul helped me fix this this morning. The issue with "Post-Build Actions.vi" was that there was an empty element in "Get UI Details.vi".